carina-utils/src/main/java/com/qaprosoft/carina/core/foundation/utils/naming/TestNamingUtil.java

```diff
@@ -218,7 +226,11 @@ private static String adjustTestName(ITestNGMethod m) {
         if (m.getDescription() != null) {
         	testName = testName.replace(SpecialKeywords.METHOD_DESCRIPTION, m.getDescription());
         } else {
-        	testName = testName.replace(SpecialKeywords.METHOD_DESCRIPTION, "");
+            if (testName.contains(SpecialKeywords.METHOD_NAME)) {
+                testName = testName.replace(SpecialKeywords.METHOD_DESCRIPTION, "");
+            } else {
+                testName = testName.replace(SpecialKeywords.METHOD_DESCRIPTION, m.getMethodName());
```

Comment:
why method_description is replaced by MethodName?

Reply:
The reason on this one, is not all test's have a description annotation by default and would cause test results to come back with a blank name.  So what we do in that situation is default back to just the standard {method_name} version.